### PR TITLE
feat: ✨ new SearchField

### PR DESCRIPTION
**Pull request by Roaa94**
Adds new `SearchField`

### CODE REVIEW

File: packages/widgetbook_core/pubspec.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ dependencies:
     sdk: flutter
 
 dev_dependencies:
+  flutter_lints: ^2.0.0
```
Comment (jenshor) [inline packages/widgetbook_core/pubspec.yaml:19]: we don't need this

File: packages/widgetbook_core/test/src/search/widgets/search_field_test.dart [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          child: SearchField(
+            onSearchPressed: voidCallbackMock,
+          ),
+        );
+        final searchButtonFinder = find.byWidgetPredicate(
+          (widget) => widget is Icon && widget.icon == Icons.search,
+        );
+
+        await tester.tap(searchButtonFinder);
+        verify(voidCallbackMock.call).called(1);
```
Comment (jenshor) [inline packages/widgetbook_core/test/src/search/widgets/search_field_test.dart:85]: nice 🎉

File: packages/widgetbook_core/assets/fonts/OFL.txt [hunk 3]
```diff
@@ -1,93 +1,93 @@
...
-COPYRIGHT HOLDER BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
-INCLUDING ANY GENERAL, SPECIAL, INDIRECT, INCIDENTAL, OR CONSEQUENTIAL
-DAMAGES, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-FROM, OUT OF THE USE OR INABILITY TO USE THE FONT SOFTWARE OR FROM
-OTHER DEALINGS IN THE FONT SOFTWARE.
+Copyright 2020 The Poppins Project Authors (https://github.com/itfoundry/Poppins)
+
+This Font Software is licensed under the SIL Open Font License, Version 1.1.
+This license is copied below, and is also available with a FAQ at:
+http://scripts.sil.org/OFL
```
Comment (jenshor) [inline packages/widgetbook_core/assets/fonts/OFL.txt:5]: Not sure why this changed but you already communicated that you're working on it.

**Review by jenshor — COMMENTED** [review]
Looks good in general and can be merged once the comments are fixed.

File: packages/widgetbook_core/pubspec.yaml (hunk 1 above)
Comment (Roaa94) [reply packages/widgetbook_core/pubspec.yaml]: Removed 👍

File: packages/widgetbook_core/assets/fonts/OFL.txt (hunk 3 above)
Comment (Roaa94) [reply packages/widgetbook_core/assets/fonts/OFL.txt]: Resolved

**Review by jenshor — APPROVED** [review]
LGTM 🎉

**Pull request merged**